File: groups/SpecialGroups.js
const text = "João é calmo, mas no transito fica nervoso."

console.log(text.match(/[\wÀ-ú]+/gi))

//Positive lookahead
console.log(text.match(/[\wÀ-ú]+(?=,)/gi))
console.log(text.match(/[\wÀ-ú]+(?=\.)/gi))
console.log(text.match(/[\wÀ-ú]+(?=\, mas)/gi))

//Negative lookahead
console.log(text.match(/[\wÀ-ú]+\b(?!,)/gi))
console.log(text.match(/[\wÀ-ú]+[\s|\.](?!,)/gi))

File: groups/SpecialGroups.rb
text = 'supermercado superação hiperMERCADO'

puts text.scan(/(?:super)[\wÀ-ú]/i).join(' ')

#Positive Lookbehind
puts text.scan(/(?<=super)[\wÀ-ú]+/i).join(' ')

#Negative Lookbehind
puts text.scan(/(?<!super)mercado/i).join(' ')